src/main/java/org/opentripplanner/apis/gtfs/datafetchers/LegImpl.java

```diff
@@ -324,4 +325,27 @@ public DataFetcher<Iterable<Leg>> nextLegs() {
   public DataFetcher<Double> accessibilityScore() {
     return environment -> NumberMapper.toDouble(getSource(environment).accessibilityScore());
   }
+
+  @Override
+  public DataFetcher<Relay.ResolvedGlobalId> id() {
+    return environment -> {
+      var leg = getSource(environment);
+      var ref = getSource(environment).getLegReference();
+      String id;
+
+      if (ref == null) {
+        id =
+          String.join(
+            "",
+            leg.start().time().toString(),
+            leg.end().time().toString(),
+            leg.getFrom().toStringShort(),
+            leg.getTo().toStringShort()
+          );
```

Comment:
Unlike the transit leg references, this is now not "double base64 encoded" (for transit the original reference is already base64 encoded and then encoded again with the `leg:` prefix). I wonder if we should encode this as well or not.

I like that this is only on the GTFS API so it doesn't potentially break transmodel clients who might rely on the fact that non-null ids are refetchable-